### PR TITLE
chore(upgrades): make compatible with Elixir 1.10

### PR DESCRIPTION
**Pull request by leifg**
[Relax dependency on plug](https://github.com/lob/litmus/pull/31/commits/eeff41fd9d95beaf508002c255c913da4e06038c) (otherwise dependent projects will always need to use 1.8)

Also update development dependencies and make tests compatible with Elixir 1.10

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -54,11 +54,11 @@ defmodule Litmus.MixProject do
...
   defp deps do
     [
-      {:ex_doc, "~> 0.20.2", only: :dev, runtime: false},
-      {:credo, "~> 1.0.5", only: [:dev, :test], runtime: false},
+      {:ex_doc, "~> 0.22", only: :dev, runtime: false},
+      {:credo, "~> 1.4", only: [:dev, :test], runtime: false},
       {:excoveralls, "~> 0.11", only: :test},
-      {:plug, "~> 1.8.0", optional: true},
-      {:dialyxir, "~> 0.5", only: [:dev, :test], runtime: false}
+      {:plug, ">= 1.8.0", optional: true},
```
Comment (mpiercy827) [inline mix.exs]: Could we make this `~> 1.8` instead?

**Review by mpiercy827 — APPROVED** [review]
LGTM, just one nit.

File: mix.exs (hunk 1 above)
Comment (leifg) [reply mix.exs]: Forget what I said, doesn't apply here. However `~> 1.8` will create another issue, will investigate.

~~Unfortunately we cannot.~~

~~Requirements with greater equals always need to be specific: https://hexdocs.pm/elixir/Version.html#module-requirements~~

Comment (leifg) [reply mix.exs]: Turns out `~> 1.8` does work

**Review by shraddha1704 — APPROVED** [review]

**Pull request merged**